test/framework/src/main/java/org/opensearch/test/OpenSearchTestClusterRule.java

```diff
@@ -56,6 +56,7 @@
  * on the way cluster settings are being managed.
  */
 class OpenSearchTestClusterRule implements MethodRule {
+    private final Map<TestCluster, OpenSearchIntegTestCase> suites = new IdentityHashMap<>();
     private final Map<Class<?>, TestCluster> clusters = new IdentityHashMap<>();
```

Comment:
Could we accomplish the same thing if the value of the `clusters` map held the TestCluster and the test instance that created it? Something like:

```java
private static class TestClusterHolder {
    // The test cluster
    TestCluster cluster;

     // The test instance that triggered the creation of this TestCluster
    OpenSearchIntegTestCase originatingTestInstance;
}
```

That might be more clear. What do you think?

Reply:
Yeah we could this as well, I didn't want to deal with wrappers to be fair because it would impact more places

Reply:
That's fair. Maybe just add a comment explaining this map:

"Maps each TestCluster to the exact test instance that triggers its creation"

Reply:
Sure, that would be helpful (sorry signed off from laptop for today , will do tomorrow morning first thing), thanks!

Reply:
Ok, got to it earlier :D should be all set now, thanks @andrross !